src/Plugin/resource/Resource.php

```diff
@@ -567,4 +568,48 @@ protected function initAuthenticationManager() {
     }
   }
 
+  /**
+   * Adds the Allowed-Origin headers.
+   *
+   * @param string $path
+   *   The requested path.
+   */
+  protected function preflight($path) {
+    $plugin_definition = $this->getPluginDefinition();
+    $allowed_origin = empty($plugin_definition['allowOrigin']) ? variable_get('restful_allowed_origin', NULL) : $plugin_definition['allow_origin'];
+    // Always add the allow origin if configured.
+    if ($allowed_origin) {
+      restful()
+        ->getResponse()
+        ->getHeaders()
+        ->add(HttpHeader::create('Access-Control-Allow-Origin', check_plain($allowed_origin)));
```

Comment:
It's not harmful, I'm just trying to be careful about user input data. To your point, I don't have an attack vector in mind.